### PR TITLE
chore(utxo-lib): port networks to typescript

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Issue: BG-7478

f8eb4df0 (Otto Allmendinger, 28 minutes ago)
refactor(utxo-lib): deprecate use of `network.coin`

We already have `getNetworkName()` and don't need three-letter codes.

Issue: BG-7478

### CODE REVIEW

File: modules/utxo-lib/test/integration_local_rpc/generate/main.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ async function initBlockchain(rpc: RpcClient, network: Network): Promise<void> {
   }
 }
 
-function toRegtestAddress(network: Network, scriptType: ScriptType, script: Buffer): string {
+function toRegtestAddress(network: { bech32?: string }, scriptType: ScriptType, script: Buffer): string {
```
Comment (sangaman) [inline modules/utxo-lib/test/integration_local_rpc/generate/main.ts:39]: Is the `bech32` variable being used anywhere?

File: modules/utxo-lib/src/networkTypes.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+export type Network = {
+  messagePrefix: string;
+  pubKeyHash: number;
+  scriptHash: number;
+  wif: number;
+  bip32: {
+    public: number;
+    private: number;
+  };
+  bech32?: string;
```
Comment (sangaman) [inline modules/utxo-lib/src/networkTypes.ts:48]: Is this supposed to represent the bech32 prefix? Just not clear what it's being used for from the changes afaict.

**Review by sangaman — APPROVED** [review]
Looks good, `networks.ts` and `networkTypes.ts` would ideally be kept around even after wrapping bitcoinjs-lib right?

File: modules/utxo-lib/test/integration_local_rpc/generate/main.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/utxo-lib/test/integration_local_rpc/generate/main.ts:39]: yep it is passed to `fromOutputScript` later

**Comment by OttoAllmendinger** [comment]
yes this code will stick around after we convert the lib to a wrapper

**Pull request merged**